api/src/main/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -257,12 +263,30 @@ public Long getDomainId() {
                 }
             }
         }
+        if(getBootType() != null){ // export to get
+            if(getBootType().equalsIgnoreCase("UEFI")) {
+                customparameterMap.put("UEFI", getBootMode());
+            }
+        }
+
         if (rootdisksize != null && !customparameterMap.containsKey("rootdisksize")) {
             customparameterMap.put("rootdisksize", rootdisksize.toString());
         }
         return customparameterMap;
     }
 
+    public String getBootType() {
+        if (bootType != null && !("bios".equalsIgnoreCase(bootType) || "uefi".equalsIgnoreCase(bootType))) {
```

Comment:
May I suggest boot type being an enum?

Reply:
Sure Enum can be integrated, will work on it.

Reply:
Changes done as suggested.